zerver/tests/test_decorators.py

```diff
@@ -389,3 +416,88 @@ def test_webhook_deactivated_user(self):
+
+        with self.assertRaises(JsonableError):
+            validate_api_key(self.webhook_bot.email, self.default_bot.api_key)
+
+    def test_validate_api_key_if_profile_is_not_active(self):
+        self._change_is_active_field(self.default_bot, False)
+        with self.assertRaises(JsonableError):
+            validate_api_key(self.default_bot.email, self.default_bot.api_key)
+        self._change_is_active_field(self.default_bot, True)
+
+    def test_validate_api_key_if_profile_is_incoming_webhook_and_is_webhook_is_unset(self):
+        with self.assertRaises(JsonableError):
+            validate_api_key(self.webhook_bot.email, self.webhook_bot.api_key)
+
+    def test_validate_api_key_if_profile_is_incoming_webhook_and_is_webhook_is_set(self):
+        profile = validate_api_key(self.webhook_bot.email, self.webhook_bot.api_key, is_webhook=True)
+        self.assertEqual(profile.pk, self.webhook_bot.pk)
+
+    def _change_is_active_field(self, profile, value):
+        profile.is_active = value
+        profile.save()
+
+
+class TestAuthenticatedJsonPostViewDecorator(AuthedTestCase):
+    def test_authenticated_json_post_view_if_everything_is_correct(self):
+        user_email = 'hamlet@zulip.com'
+        self._login(user_email)
+        response = self._do_test(user_email)
+        self.assertEqual(response.status_code, 200)
+
+    def test_authenticated_json_post_view_if_user_is_incoming_webhook(self):
+        user_email = 'webhook-bot@zulip.com'
+        self._login(user_email)
+        self.assertJsonableError(self._do_test(user_email), "Webhook bots can only access webhooks")
+
+    def test_authenticated_json_post_view_if_user_is_not_active(self):
+        user_email = 'hamlet@zulip.com'
+        user_profile = get_user_profile_by_email(user_email)
+        old_is_active_value = user_profile.is_active
+        self._login(user_email)
+        self._change_is_active_user_value(user_profile, False)
+        self.assertJsonableError(self._do_test(user_email), "Account not active")
+        self._change_is_active_user_value(user_profile, old_is_active_value)
+
+    def test_authenticated_json_post_view_if_user_realm_is_deactivated(self):
+        user_email = 'hamlet@zulip.com'
+        user_profile = get_user_profile_by_email(user_email)
+        old_is_realm_deactivated_value = user_profile.realm.deactivated
+        self._login(user_email)
+        self._change_realm_deactivated_value(user_profile, True)
+        self.assertJsonableError(self._do_test(user_email), "Realm for account has been deactivated")
+        self._change_realm_deactivated_value(user_profile, old_is_realm_deactivated_value)
+
+    def _change_realm_deactivated_value(self, user_profile, value):
+        user_profile.realm.deactivated = value
+        user_profile.save()
+
+    def _change_is_active_user_value(self, user_profile, value):
+        user_profile.is_active = value
+        user_profile.save()
```

Comment:
you can use `do_deactivate_user` and `do_reactivate_user` for these (and similarly for the realm ones); the other tests in this file already use those functions, so I think you should do the same here.